### DISCUSSION TITLE
Use with Headless WordPress as backend

### DISCUSSION THREAD

**Discussion by ShahriarKh** [nextauthjs/next-auth#3927] [Help]
Is it possible to use a headless WordPress + GraphQL as a backend that handles sign-up, login, ...?
Currently, I'm using WordPress as a headless cms for my Next.js project. I also found it possible to handle login, sign up and most of user-related things with plugins like [wp-graphQL](https://www.wpgraphql.com/) and its extensions like https://github.com/wp-graphql/wp-graphql-jwt-authentication.
However, I'm just getting started with NextAuth and have no idea how should I connect them together.
Do you have any tutorials, guides, or tips?
Any help is highly appreciated 😺

**Comment by ariburaco** [comment]
I am wondering about this feature also. Is there any update to use nextauth with headless wordpress + grahpql?

**Comment by stephane-segning** [comment]
Hello, any new updates about this ?

**Comment by ChenemiAbraham** [comment]
We are all looking to solve this problem. Can we join hands together to build a plugin that solves this?

**Comment by cedricrabarijohn** [comment]
The only way I found was to create a fully customized adapter for it, therefore I don't know if I'm doing it in the "good" way, since it's a bunch of CRUD manually written and served via wpGraphql. 

The model I followed https://authjs.dev/getting-started/database#models
